### PR TITLE
Revamp model parallel docs (1/n)

### PR DESCRIPTION
**Pull request by awaelchli**
## What does this PR do?

Part of #18083

This PR moves FSDP, DeepSpeed, DDP optimizations and efficient model init to separate pages, that were previously in one humongous page that has grown too big. No content changes in this PR. Each of these pages will be overhauled in follow-up PRs. 

https://pytorch-lightning--18314.org.readthedocs.build/en/18314/levels/advanced.html

cc @borda @awaelchli @carmocca

### CODE REVIEW

File: docs/source-pytorch/advanced/model_parallel/index.rst [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+For training on unreliable mixed GPUs across the internet check out the :doc:`Hivemind Strategy <../../integrations/strategies/hivemind>` integration.
+
+
+----
+
+
+******
+Guides
+******
```
Comment (awaelchli) [inline docs/source-pytorch/advanced/model_parallel/index.rst:73]: This section is just a placeholder for navigation right now. The entire page here needs an overhaul.

File: docs/source-pytorch/accelerators/gpu_advanced.rst [hunk 2]
```diff
@@ -22,7 +22,7 @@ For experts pushing the state-of-the-art in model development, Lightning offers
    :header: Train 1 trillion+ parameter models
    :description:
    :col_css: col-md-4
-   :button_link: ../advanced/model_parallel.html
+   :button_link: ../advanced/model_parallel/index.html
```
Comment (Borda) [inline docs/source-pytorch/accelerators/gpu_advanced.rst:25]: let's add some compatibility redirects

File: docs/source-pytorch/advanced/model_parallel/index.rst (hunk 1 above)
Comment (carmocca) [reply docs/source-pytorch/advanced/model_parallel/index.rst:73]: Shouldn't you also link to ddp_optimizations and model_init here for now?

**Review by carmocca — APPROVED** [review]
Thanks for starting this

File: docs/source-pytorch/accelerators/gpu_advanced.rst (hunk 2 above)
Comment (awaelchli) [reply docs/source-pytorch/accelerators/gpu_advanced.rst:25]: It would be great to do this in a separate PR :)

**Review by Borda — APPROVED** [review]

File: docs/source-pytorch/advanced/model_parallel/index.rst (hunk 1 above)
Comment (awaelchli) [reply docs/source-pytorch/advanced/model_parallel/index.rst]: Yes, these will be nicely linked. Follow-up incoming shortly.

**Pull request merged**